project-koku/nise-populator
# COST-2381: Add OCP on GCP to nise populator flow.

Adding OCP on GCP to nise populator flow. 

## How to test locally:
1.  Add new env vars:
You will need to fill out these .env vars:
```
GCP_DATASET=GCP_DATASET
GCP_PROJECT_ID=GCP_PROJECT_ID
GOOGLE_APPLICATION_CREDENTIALS=/path/to/creds.json
```
Ask me if you don't know these values.

2. If you are like me and don't have info for the other providers you will need temporary tell source factor to only run for GCP:
```
self.valid_sources = [AWS, Azure, OCP, GCP]
```
To:
```
self.valid_sources = [GCP]
```

3. Run app.py
```
python3 nise-populator/app.py
```
You should see something like this in the logs:
```
2022-03-29 12:52:09,248 - nise.util.log - INFO - Dataset FAKE_DATASET created in GCP bigquery under the table name FAKE_TABLE_NAME_ETAG.
```
